### PR TITLE
Format `.swcrc` as a JSON file

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

SWC is a blazing fast JavaScript transpiler written with Rust.

`.swcrc` is a JSON file that configures SWC. Please see https://swc.rs/docs/configuration/swcrc for more details.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/integration/__tests__/with-parser-inference.js [hunk 1]
```diff
@@ -45,4 +45,10 @@ describe("infers parser from filename", () => {
       prettier.format("let foo = ( x = 1 ) => x", { filepath: "x/y/Jakefile" })
     ).toBe("let foo = (x = 1) => x;\n");
   });
+
+  test("json from .swcrc", () => {
+    expect(prettier.format("  {   }  ", { filepath: "x/y/.swcrc" })).toBe(
```
[the remark this answers is not in the document]
Comment (fisker) [reply tests/integration/__tests__/with-parser-inference.js:50]: I thought it's a regexp at first.

[the remark this answers is not in the document]
Comment (fisker) [reply tests/integration/__tests__/with-parser-inference.js:50]: Maybe good to add comments inside.

[the remark this answers is not in the document]
Comment (sosukesuzuki) [reply tests/integration/__tests__/with-parser-inference.js]: What should we comment?

[the remark this answers is not in the document]
Comment (fisker) [reply tests/integration/__tests__/with-parser-inference.js]: I mean in json

```
expect(prettier.format(`
{
  "jsc": {
// Requires v1.2.50 or upper and requires target to be es2016 or upper.
    "keepClassNames": false
  }
}
`, { filepath: "/path/to/.swcrc" })).toBe(
```

**Review by kachkaev — APPROVED** [review]

**Pull request merged**